src/vtp/web/api/main.py

```diff
@@ -22,19 +41,63 @@ async def root() -> dict:
     return {"version": "0.1.0"}
 
 
+# Endpoint #1
 @app.get("/vote/")
 async def get_vote_store_id() -> dict:
     """Create and store a unique Vote Store ID for each client"""
-    vote_store_id = str(random.randrange(100000, 999999))
+
+    svdo = SetupVtpDemoOperation(
+        election_data_dir=DEFAULT_ELECTION_DATA_DIR,
+        verbosity=VERBOSITY,
+    )
+    guid = svdo.run(
+        guid_client_store=True,
+    )
+
+    vote_store_id = guid
     # add VoteStoreID to list
     vote_store_ids.append(vote_store_id)
     return {"VoteStoreID": vote_store_id}
 
 
+# Endpoint #2
 @app.get("/vote/{vote_store_id}")
 async def get_empty_ballot(vote_store_id: str) -> dict:
     """Return an empty ballot for a given Vote Store ID"""
+
+    if vote_store_id in vote_store_ids:
+        # Cet a (the) blank ballot from the backend
+        cbo = CastBallotOperation(
+            guid=vote_store_id,
+            verbosity=VERBOSITY,
+        )
+        empty_ballot = cbo.run(
+            blank_ballot=DEFAULT_BLANK_BALLOT,
+            return_bb=True,
+        )
+        return {"blank-ballot": empty_ballot}
+    return {"error": "VoteStoreID not found"}
+
+
+# Endpoint #3
+@app.get("/vote/cast-ballot/{vote_store_id}")
+async def cast_ballot(vote_store_id: str, cast_ballot_json: dict = None) -> dict:
+    """
+    Uploads the ballot; the backend accepts it, merges it, and returns
+    the ballot check and voter index
+    """
+
     if vote_store_id in vote_store_ids:
-        return {"ballot": f"{empty_ballot}"}
-    else:
-        return {"error": "VoteStoreID not found"}
+        # Accept the incoming cast ballot json and return a ballot check and voter index
+        abo = AcceptBallotOperation(
+            guid=vote_store_id,
+            verbosity=VERBOSITY,
+        )
+        # ZZZ stub out cast ballot
+        ballot_check, voter_index = abo.run(
+            cast_ballot=DEFAULT_CAST_BALLOT,
+            cast_ballot_json=cast_ballot_json,
+            merge_contests=True,
+        )
+        return {"ballot_check": ballot_check, "voter_index": voter_index}
+    return {"error": "VoteStoreID not found"}
```

Comment:
Assuming you'll move most of the code to that external vtp ops module, again, I'd suggest combining the if statement with the return in a one-liner, as in the previous endpoint.

Reply:
Yes.  Question, when you say an external vtp ops module, do you mean a file in this repo or a file in the VoteTrackerPlus repo?

Reply:
As noted, a file in this repo.